Source/Protected.swift

```diff
@@ -82,6 +82,45 @@ final class MutexLock: Lock {
 }
 #endif
 
+#if os(Windows)
+import WinSDK
+private func randomString(length: Int) -> String {
+    let letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+    return String((0..<length).map{ _ in letters.randomElement()! })
+}
+
+/// A `CreateMutexA` wrapper.
+final class MutexLock: Lock {
```

Comment:
I may be better, at least for this initial version, to wrap `NSLock` on Windows.  I don't understand the Windows locking primitives at all, so it would probably be better to use an implementation that has already been proven.

Reply:
Done

Reply:
Sorry, now that I think about it, what about just making `NSLock` conform to our `Lock` protocol and using it directly?

Reply:
That’s probably better, do you think we even need the apple OS implementation of `UnfairLock`? Just add an extension to NSLock. This on its own compiles:
```
extension NSLock: Lock { }
```
because NSLock already has `lock` and `unlock` methods?

Reply:
Yeah, then we can use it on the other platforms. We should keep `UnfairLock` since it's much faster on Apple platforms.

Reply:
Got it, pushed